.github/workflows/deploy.yml

```diff
@@ -1,21 +1,31 @@
 name: Deploy
 
 on:
-  workflow_dispatch:
+  release:
+    types: [published]
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - name: Setup .NET
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 6.0.x
+          dotnet-version: |
+            3.1.x
+            6.0.x
+            7.0.x
+      - name: Build
+        run: ./build.sh --task=Build --configuration=Release
```

Comment:
@seamasmcgettrick Out of curiosity, why do you use cake here (`./build.sh ...`) but use `dotnet` directly in the `test.yml` workflow?